lib/utils.js

```diff
@@ -36,8 +36,11 @@ function InvalidAlgorithmError(message) {
 util.inherits(InvalidAlgorithmError, HttpSignatureError);
 
 function validateAlgorithm(algorithm) {
-  var alg = algorithm.toLowerCase().split('-');
+  if (algorithm.toLowerCase() === 'hs2019') {
+    return ['hmac', 'sha512'];
```

Comment:
I believe in case of `hs2019` you shouldn't assume that hmac will be used, but pick the algorithm based on what key was provided locally.
See here: https://github.com/voltone/plug_signature/blob/989a7e8d347912a14e721f96a42a27aaf054c724/lib/plug_signature/crypto.ex

If we wan't to publish this lib, we should make it more universal, but such shortcut is probably ok for our internal use.